pkgs/by-name/me/mealie/package.nix

```diff
@@ -0,0 +1,111 @@
+{ lib, callPackage, fetchFromGitHub, nixosTests, python3, python3Packages, writeShellScript }:
+
+let
+  version = "1.1.0";
+  src = fetchFromGitHub {
+    owner = "mealie-recipes";
+    repo = "mealie";
+    rev = "v${version}";
+    sha256 = "sha256-k2jMGJ6bdhkXootWl9cEdevRrGfggmHbkJZHZmhNLTM=";
+  };
+
+  frontend = callPackage (import ./mealie-frontend.nix src version) { };
+
+  # TODO  Add crfpp in runtime dependencies for ingredient quantities NLP
+
+in python3Packages.buildPythonPackage rec {
+  pname = "mealie";
+  inherit version src;
+  format = "pyproject";
+
+  patches = [
+    # Replace hardcoded paths in code with environment variables (meant for inside Docker only)
+    # So we can configure easily where the data is stored on the server
+    ./mealie_statedir.patch
+    ./mealie_init_db.patch
+    ./mealie_logger.patch
+  ];
+
+  nativeBuildInputs = [
+    python3Packages.poetry-core
+    python3Packages.pythonRelaxDepsHook
+  ];
+
+  pythonRelaxDeps = true;
+
+  propagatedBuildInputs = with python3Packages; [
+    aiofiles
+    alembic
+    aniso8601
+    appdirs
+    apprise
+    bcrypt
+    extruct
+    (fastapi.override {
+      pydantic = pydantic_1;
+    })
+    gunicorn
+    html2text
+    jinja2
+    lxml
+    orjson
+    passlib
+    pillow
+    psycopg2
+    pyhumps
+    pytesseract
+    python-dotenv
+    python-jose
+    python-ldap
+    python-multipart
+    python-slugify
+    pyyaml
+    rapidfuzz
+    recipe-scrapers
+    sqlalchemy
+    tzdata
+    uvicorn
+  ];
```

Comment:
From what I've seen, the usual way to go about overriding some of the python packages for a `buildPythonPackage` is to use `packageOverrides`, see [`tandoor-recipes`](https://github.com/NixOS/nixpkgs/blob/f4a8d6d5324c327dcc2d863eb7f3cc06ad630df4/pkgs/applications/misc/tandoor-recipes/default.nix#L8) for example.

So something like the following should work:

```nix
  python = python3.override {
    packageOverrides = self: super: {
      pydantic = self.pydantic_1;
    };
  };
```

And do `passthru = { inherit python; }` as well.

Reply:
Ah and of course, make sure you use `propagatedBuildInputs = with python.pkgs; [ ... ]` to use those overrides.